src/pages/create-course/CreateCourse.tsx

```diff
@@ -10,12 +11,62 @@ import { authRoutes } from '~/router/constants/authRoutes'
 import { ButtonTypeEnum, ButtonVariantEnum } from '~/types'
 
 import { styles } from '~/pages/create-course/CreateCourse.styles'
+import { countActiveOfferFilters } from '~/utils/count-active-filters'
+import { useFilterQuery } from '~/hooks/use-filter-query'
+import { useAppSelector } from '~/hooks/use-redux'
+import { getOpositeRole } from '~/utils/helper-functions'
+import { defaultFilters } from '~/pages/find-offers/FindOffers.constants'
+
+import CourseSearchToolbar from '~/containers/my-courses/course-search-tool-bar/CourseSearchToolbar'
+import CourseCreateDescription from '~/containers/my-courses/course-search-tool-bar/CourseCreateDescription'
 
 const CreateCourse = () => {
   const { t } = useTranslation()
   const navigate = useNavigate()
+
+  const { userRole } = useAppSelector((state) => state.appMain)
+
+  const oppositeRole = getOpositeRole(userRole)
+
+  const { filters, filterQueryActions } = useFilterQuery({
+    defaultFilters: defaultFilters(oppositeRole),
+    countActiveFilters: countActiveOfferFilters
+  })
+
+  const resetPage = () => {
+    filterQueryActions.updateFilterInQuery(
+      defaultFilters(oppositeRole).page,
+      'page'
+    )
+  }
+
+  const [titleName, setTitleName] = useState<string>('')
+  const [decriptionName, setDescriptionName] = useState<string>('')
+
+  const onChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
+    const newValue = e.target.value
+    setTitleName(newValue)
+  }
+  const onChangeHandlerDesc = (e: ChangeEvent<HTMLInputElement>) => {
+    const newValue = e.target.value
+    setDescriptionName(newValue)
```

Comment:
hmm, do we really need to create a new variable just to set it in state?
maybe we can do it just like this?
```js
const onChangeHandler = (e: ChangeEvent<HTMLInputElement>) => setTitleName(e.target.value)

const onChangeHandlerDesc = (e: ChangeEvent<HTMLInputElement>) => setDescriptionName(e.target.value)
```

Reply:
done